pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarInternalColumn.java

```diff
@@ -39,6 +40,18 @@
 
 public abstract class PulsarInternalColumn {
 
+    public static class PartitionColumn extends PulsarInternalColumn {
+
+        PartitionColumn(String name, Type type, String comment) {
+            super(name, type, comment);
+        }
+
+        @Override
+        public Object getData(RawMessage message) {
```

Comment:
You can get `partition` from the messageId, no?

Reply:
RawMessage does not contains partition now and get partition by split will reduce memory usage more than storing a partition per RawMessage.